Add enterprise events endpoint

emilykoh opened the pull request:
Adding the functionality needed to support the new enterprise events endpoint

* Ready for review but not for merging
* Added functional code walkthrough in Advanced readme
* ~~Paolo said that there might be upcoming changes with the new release of the enterprise events, so I'm waiting to see if any changes will be needed for the ruby SDK before merging this PR.~~

PauloLisboa commented:
Emily,
I think we need better functional tests so that users can look at the tests
to find samples on how to call* Events API*.
I noticed that the javascript SDK has a few nice functional tests for
*Sheets* API. I didn't see similar tests for *Events* API.
Given that Events API is just one call to GET /events?... then I would ask
you to have samples using each one (combined or in isolation) of the
supported querystring parameters (i.e. *since*, *streamPosition*, *maxCount*,
*numericDates*).

BTW, you probably noticed that *GET /events *call fails if it doesn't have
either *since* or *streamPosition *passed in the querystring (because GET
events needs to know the start point for the batch of events you are
requesting).

I suggest the sample functional test below to cover lots of scenarios:

1) Call the Events API with querystring parameters
*since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
2) Assign the response to variable *previousResponse*
3) Display on the console the items in *previousResponse.data*
4) Start a *while loop* where the criteria to keep running is
*previousResponse.moreAvailable* must be true
5) Set to variable *nextStreamPosition* the value in
*previousResponse.nextStreamPosition*
6) Call Events API with querystring* streamPosition=<variable
nextStreamPosition>&numericDates=false&maxCount=10*
7) Assign the response to variable *previousResponse*
8) Display on the console the items in *previousResponse.data*
9) Repeate loop

Thank you,

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com




On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <notifications@github.com>
wrote:

> @kfansler <https://github.com/kfansler> requested your review on: #76
> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76>
> Add enterprise events endpoint .
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d>
> .
>

PauloLisboa commented:
BTW, event though I had already approved the Ruby SDK, we need the same
thing (better Functional test) for the Ruby SDK too; mainly that sample
with while loop that I suggested for javascript because that clarifies how
to iterate though the batches to get all the data.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com




On Fri, Apr 12, 2019 at 4:41 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
wrote:

> Emily,
> I think we need better functional tests so that users can look at the
> tests to find samples on how to call* Events API*.
> I noticed that the javascript SDK has a few nice functional tests for
> *Sheets* API. I didn't see similar tests for *Events* API.
> Given that Events API is just one call to GET /events?... then I would ask
> you to have samples using each one (combined or in isolation) of the
> supported querystring parameters (i.e. *since*, *streamPosition*,
> *maxCount*, *numericDates*).
>
> BTW, you probably noticed that *GET /events *call fails if it doesn't
> have either *since* or *streamPosition *passed in the querystring
> (because GET events needs to know the start point for the batch of events
> you are requesting).
>
> I suggest the sample functional test below to cover lots of scenarios:
>
> 1) Call the Events API with querystring parameters
> *since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
> 2) Assign the response to variable *previousResponse*
> 3) Display on the console the items in *previousResponse.data*
> 4) Start a *while loop* where the criteria to keep running is
> *previousResponse.moreAvailable* must be true
> 5) Set to variable *nextStreamPosition* the value in
> *previousResponse.nextStreamPosition*
> 6) Call Events API with querystring* streamPosition=<variable
> nextStreamPosition>&numericDates=false&maxCount=10*
> 7) Assign the response to variable *previousResponse*
> 8) Display on the console the items in *previousResponse.data*
> 9) Repeate loop
>
> Thank you,
>
> Paulo Lisboa
> Senior Software Development Engineer
> paulo.lisboa@smartsheet.com
>
>
>
>
> On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <notifications@github.com>
> wrote:
>
>> @kfansler <https://github.com/kfansler> requested your review on: #76
>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76>
>> Add enterprise events endpoint .
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d>
>> .
>>
>

PauloLisboa commented:
Kevin,
I changed the subject and explicitly included you, Emily, and Tim here to
make sure it reached you.
My test/sample suggestion in the email below applies to all languages.

Thanks.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com




On Fri, Apr 12, 2019 at 5:24 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
wrote:

>
> BTW, event though I had already approved the Ruby SDK, we need the same
> thing (better Functional test) for the Ruby SDK too; mainly that sample
> with while loop that I suggested for javascript because that clarifies how
> to iterate though the batches to get all the data.
>
> Paulo Lisboa
> Senior Software Development Engineer
> paulo.lisboa@smartsheet.com
>
>
>
>
> On Fri, Apr 12, 2019 at 4:41 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
> wrote:
>
>> Emily,
>> I think we need better functional tests so that users can look at the
>> tests to find samples on how to call* Events API*.
>> I noticed that the javascript SDK has a few nice functional tests for
>> *Sheets* API. I didn't see similar tests for *Events* API.
>> Given that Events API is just one call to GET /events?... then I would
>> ask you to have samples using each one (combined or in isolation) of the
>> supported querystring parameters (i.e. *since*, *streamPosition*,
>> *maxCount*, *numericDates*).
>>
>> BTW, you probably noticed that *GET /events *call fails if it doesn't
>> have either *since* or *streamPosition *passed in the querystring
>> (because GET events needs to know the start point for the batch of events
>> you are requesting).
>>
>> I suggest the sample functional test below to cover lots of scenarios:
>>
>> 1) Call the Events API with querystring parameters
>> *since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
>> 2) Assign the response to variable *previousResponse*
>> 3) Display on the console the items in *previousResponse.data*
>> 4) Start a *while loop* where the criteria to keep running is
>> *previousResponse.moreAvailable* must be true
>> 5) Set to variable *nextStreamPosition* the value in
>> *previousResponse.nextStreamPosition*
>> 6) Call Events API with querystring* streamPosition=<variable
>> nextStreamPosition>&numericDates=false&maxCount=10*
>> 7) Assign the response to variable *previousResponse*
>> 8) Display on the console the items in *previousResponse.data*
>> 9) Repeate loop
>>
>> Thank you,
>>
>> Paulo Lisboa
>> Senior Software Development Engineer
>> paulo.lisboa@smartsheet.com
>>
>>
>>
>>
>> On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <notifications@github.com>
>> wrote:
>>
>>> @kfansler <https://github.com/kfansler> requested your review on: #76
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76>
>>> Add enterprise events endpoint .
>>>
>>> —
>>> You are receiving this because your review was requested.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d>
>>> .
>>>
>>

emilykoh commented:
Paulo, I added the functional test for calling the enterprise events endpoint, and the assertions that is happening on the test is that the returned response matches the expected structure of the response. 

I did not include the console.logs because it's a test, but the test itself should give users a good idea of how the endpoint is designed to be used. 

Please let me know if there's anything weird with the way I'm using the endpoint, but I'm currently under the understanding that you would first pass in the `since` parameter with the query parameters, and then use the returned `nextStreamPosition` as the next value for accessing the endpoint.

emilykoh commented:
> Paulo, I added the functional test for calling the enterprise events endpoint, and the assertions that is happening on the test is that the returned response matches the expected structure of the response.
> 
> I did not include the console.logs because it's a test, but the test itself should give users a good idea of how the endpoint is designed to be used.
> 
> Please let me know if there's anything weird with the way I'm using the endpoint, but I'm currently under the understanding that you would first pass in the `since` parameter with the query parameters, and then use the returned `nextStreamPosition` as the next value for accessing the endpoint.

Travis CI check is currently failing, will investigate tomorrow morning.

emilykoh commented:
Hello Paulo,

I've updated the JS pull request with a functional test per your suggestion, so could you please review the test when you have the time to look over it? If it looks good then I'll add something similar to the Ruby SDK, otherwise please feel free to give me feedback on the test!

Best,
Emily
________________________________
From: Paulo Lisboa <notifications@github.com>
Sent: Monday, April 15, 2019 2:23 PM
To: smartsheet-platform/smartsheet-javascript-sdk
Cc: Koh, Emily; Author
Subject: Re: [smartsheet-platform/smartsheet-javascript-sdk] Add enterprise events endpoint (#76)

Kevin,
I changed the subject and explicitly included you, Emily, and Tim here to
make sure it reached you.
My test/sample suggestion in the email below applies to all languages.

Thanks.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com




On Fri, Apr 12, 2019 at 5:24 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
wrote:

>
> BTW, event though I had already approved the Ruby SDK, we need the same
> thing (better Functional test) for the Ruby SDK too; mainly that sample
> with while loop that I suggested for javascript because that clarifies how
> to iterate though the batches to get all the data.
>
> Paulo Lisboa
> Senior Software Development Engineer
> paulo.lisboa@smartsheet.com
>
>
>
>
> On Fri, Apr 12, 2019 at 4:41 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
> wrote:
>
>> Emily,
>> I think we need better functional tests so that users can look at the
>> tests to find samples on how to call* Events API*.
>> I noticed that the javascript SDK has a few nice functional tests for
>> *Sheets* API. I didn't see similar tests for *Events* API.
>> Given that Events API is just one call to GET /events?... then I would
>> ask you to have samples using each one (combined or in isolation) of the
>> supported querystring parameters (i.e. *since*, *streamPosition*,
>> *maxCount*, *numericDates*).
>>
>> BTW, you probably noticed that *GET /events *call fails if it doesn't
>> have either *since* or *streamPosition *passed in the querystring
>> (because GET events needs to know the start point for the batch of events
>> you are requesting).
>>
>> I suggest the sample functional test below to cover lots of scenarios:
>>
>> 1) Call the Events API with querystring parameters
>> *since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
>> 2) Assign the response to variable *previousResponse*
>> 3) Display on the console the items in *previousResponse.data*
>> 4) Start a *while loop* where the criteria to keep running is
>> *previousResponse.moreAvailable* must be true
>> 5) Set to variable *nextStreamPosition* the value in
>> *previousResponse.nextStreamPosition*
>> 6) Call Events API with querystring* streamPosition=<variable
>> nextStreamPosition>&numericDates=false&maxCount=10*
>> 7) Assign the response to variable *previousResponse*
>> 8) Display on the console the items in *previousResponse.data*
>> 9) Repeate loop
>>
>> Thank you,
>>
>> Paulo Lisboa
>> Senior Software Development Engineer
>> paulo.lisboa@smartsheet.com
>>
>>
>>
>>
>> On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <notifications@github.com>
>> wrote:
>>
>>> @kfansler <https://github.com/kfansler<https://protect-us.mimecast.com/s/qPe2CXD703tXvqx8H6F3s6>> requested your review on: #76
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76<https://protect-us.mimecast.com/s/Ko0xCYEQn4fLRNznSGpAEk>>
>>> Add enterprise events endpoint .
>>>
>>> —
>>> You are receiving this because your review was requested.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522<https://protect-us.mimecast.com/s/fTrGCZ6wo5S5ZD3kFKBdYp>>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d<https://protect-us.mimecast.com/s/VHRbC1wYn5uMyk7PhXAMIQ>>
>>> .
>>>
>>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://protect-us.mimecast.com/s/PUZ6C2kgo6spqRNBT9WAa9>, or mute the thread<https://protect-us.mimecast.com/s/3-pyC31jp7Up1G5yTvTiRC>.



**This message originated from outside Pariveda Solutions**

PauloLisboa commented on test/functional/events_test.js:
```diff
@@ -0,0 +1,53 @@
+        async function getEvents(moreEventsAvailable) {
+        do {
+            let result = await smartsheet.events.getEvents(options);
+            should(result.data).have.property('nextStreamPosition');
```
This test isn't failing when expected. For example, the test doesn't fail even if you injected the lines of code below. The test doesn't make sense to me unless it fails and succeeds when expected. I suspect (please verify) the weird behavior is due to the fact that the test function is defined as asynchronous (async), and there is no other test in the entire SDK defined as asynchronous. I suggest you eliminate the async keyword and make it synchronous as the others.
            should(result.data).have.property('XXXXXX');
            should(YYYYYYYY).have.property('data');

Also, when debugging the test, I cannot succeed to make the Events API call work even when I give a valid Access Token. I am using the following way to define the smartsheet client:
smartsheet = client.createClient({accessToken:'MyToken', logLevel:'info', baseUrl:'https://api.plisboa.smart.ninja/2.0/'});

Please validate the test actually works when you give a valid Token and baseUrl but please show me the code that actually makes it work.

emilykoh commented:
Thank you Emily, I finished this review and submitted my comments.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com




On Mon, Apr 22, 2019 at 10:16 AM Koh, Emily <Emily.koh@parivedasolutions.com>
wrote:

> Hello Paulo,
>
> I've updated the JS pull request with a functional test per your
> suggestion, so could you please review the test when you have the time to
> look over it? If it looks good then I'll add something similar to the Ruby
> SDK, otherwise please feel free to give me feedback on the test!
>
> Best,
> Emily
> ------------------------------
> *From:* Paulo Lisboa <notifications@github.com>
> *Sent:* Monday, April 15, 2019 2:23 PM
> *To:* smartsheet-platform/smartsheet-javascript-sdk
> *Cc:* Koh, Emily; Author
> *Subject:* Re: [smartsheet-platform/smartsheet-javascript-sdk] Add
> enterprise events endpoint (#76)
>
> Kevin,
> I changed the subject and explicitly included you, Emily, and Tim here to
> make sure it reached you.
> My test/sample suggestion in the email below applies to all languages.
>
> Thanks.
>
> Paulo Lisboa
> Senior Software Development Engineer
> paulo.lisboa@smartsheet.com
>
>
>
>
> On Fri, Apr 12, 2019 at 5:24 PM Paulo Lisboa <paulo.lisboa@smartsheet.com>
> wrote:
>
> >
> > BTW, event though I had already approved the Ruby SDK, we need the same
> > thing (better Functional test) for the Ruby SDK too; mainly that sample
> > with while loop that I suggested for javascript because that clarifies
> how
> > to iterate though the batches to get all the data.
> >
> > Paulo Lisboa
> > Senior Software Development Engineer
> > paulo.lisboa@smartsheet.com
> >
> >
> >
> >
> > On Fri, Apr 12, 2019 at 4:41 PM Paulo Lisboa <
> paulo.lisboa@smartsheet.com>
> > wrote:
> >
> >> Emily,
> >> I think we need better functional tests so that users can look at the
> >> tests to find samples on how to call* Events API*.
> >> I noticed that the javascript SDK has a few nice functional tests for
> >> *Sheets* API. I didn't see similar tests for *Events* API.
> >> Given that Events API is just one call to GET /events?... then I would
> >> ask you to have samples using each one (combined or in isolation) of the
> >> supported querystring parameters (i.e. *since*, *streamPosition*,
> >> *maxCount*, *numericDates*).
> >>
> >> BTW, you probably noticed that *GET /events *call fails if it doesn't
> >> have either *since* or *streamPosition *passed in the querystring
> >> (because GET events needs to know the start point for the batch of
> events
> >> you are requesting).
> >>
> >> I suggest the sample functional test below to cover lots of scenarios:
> >>
> >> 1) Call the Events API with querystring parameters
> >> *since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
> >> 2) Assign the response to variable *previousResponse*
> >> 3) Display on the console the items in *previousResponse.data*
> >> 4) Start a *while loop* where the criteria to keep running is
> >> *previousResponse.moreAvailable* must be true
> >> 5) Set to variable *nextStreamPosition* the value in
> >> *previousResponse.nextStreamPosition*
> >> 6) Call Events API with querystring* streamPosition=<variable
> >> nextStreamPosition>&numericDates=false&maxCount=10*
> >> 7) Assign the response to variable *previousResponse*
> >> 8) Display on the console the items in *previousResponse.data*
> >> 9) Repeate loop
> >>
> >> Thank you,
> >>
> >> Paulo Lisboa
> >> Senior Software Development Engineer
> >> paulo.lisboa@smartsheet.com
> >>
> >>
> >>
> >>
> >> On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <
> notifications@github.com>
> >> wrote:
> >>
> >>> @kfansler <https://github.com/kfansler
> <https://protect-us.mimecast.com/s/qPe2CXD703tXvqx8H6F3s6>> requested
> your review on: #76
> >>> <
> https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76
> <https://protect-us.mimecast.com/s/Ko0xCYEQn4fLRNznSGpAEk>>
> >>> Add enterprise events endpoint .
> >>>
> >>> —
> >>> You are receiving this because your review was requested.
> >>> Reply to this email directly, view it on GitHub
> >>> <
> https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522
> <https://protect-us.mimecast.com/s/fTrGCZ6wo5S5ZD3kFKBdYp>>,
> >>> or mute the thread
> >>> <
> https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d
> <https://protect-us.mimecast.com/s/VHRbC1wYn5uMyk7PhXAMIQ>>
> >>> .
> >>>
> >>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://protect-us.mimecast.com/s/PUZ6C2kgo6spqRNBT9WAa9>, or mute the
> thread <https://protect-us.mimecast.com/s/3-pyC31jp7Up1G5yTvTiRC>.
>
>
> * **This message originated from outside Pariveda Solutions** *
>

emilykoh commented:
Paulo,

Thanks for the feedback, I saw your comments and will work on addressing them this week.

Best,
Emily
________________________________
From: Paulo Lisboa <paulo.lisboa@smartsheet.com>
Sent: Wednesday, April 24, 2019 9:19 AM
To: Koh, Emily
Cc: smartsheet-platform/smartsheet-javascript-sdk; smartsheet-platform/smartsheet-javascript-sdk; Author
Subject: Re: [smartsheet-platform/smartsheet-javascript-sdk] Add enterprise events endpoint (#76)


Thank you Emily, I finished this review and submitted my comments.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>

[https://www.smartsheet.com/sites/default/files/Email-Signature-Simple-Logo-2.png]


On Mon, Apr 22, 2019 at 10:16 AM Koh, Emily <Emily.koh@parivedasolutions.com<mailto:Emily.koh@parivedasolutions.com>> wrote:
Hello Paulo,

I've updated the JS pull request with a functional test per your suggestion, so could you please review the test when you have the time to look over it? If it looks good then I'll add something similar to the Ruby SDK, otherwise please feel free to give me feedback on the test!

Best,
Emily
________________________________
From: Paulo Lisboa <notifications@github.com<mailto:notifications@github.com>>
Sent: Monday, April 15, 2019 2:23 PM
To: smartsheet-platform/smartsheet-javascript-sdk
Cc: Koh, Emily; Author
Subject: Re: [smartsheet-platform/smartsheet-javascript-sdk] Add enterprise events endpoint (#76)

Kevin,
I changed the subject and explicitly included you, Emily, and Tim here to
make sure it reached you.
My test/sample suggestion in the email below applies to all languages.

Thanks.

Paulo Lisboa
Senior Software Development Engineer
paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>




On Fri, Apr 12, 2019 at 5:24 PM Paulo Lisboa <paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>>
wrote:

>
> BTW, event though I had already approved the Ruby SDK, we need the same
> thing (better Functional test) for the Ruby SDK too; mainly that sample
> with while loop that I suggested for javascript because that clarifies how
> to iterate though the batches to get all the data.
>
> Paulo Lisboa
> Senior Software Development Engineer
> paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>
>
>
>
>
> On Fri, Apr 12, 2019 at 4:41 PM Paulo Lisboa <paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>>
> wrote:
>
>> Emily,
>> I think we need better functional tests so that users can look at the
>> tests to find samples on how to call* Events API*.
>> I noticed that the javascript SDK has a few nice functional tests for
>> *Sheets* API. I didn't see similar tests for *Events* API.
>> Given that Events API is just one call to GET /events?... then I would
>> ask you to have samples using each one (combined or in isolation) of the
>> supported querystring parameters (i.e. *since*, *streamPosition*,
>> *maxCount*, *numericDates*).
>>
>> BTW, you probably noticed that *GET /events *call fails if it doesn't
>> have either *since* or *streamPosition *passed in the querystring
>> (because GET events needs to know the start point for the batch of events
>> you are requesting).
>>
>> I suggest the sample functional test below to cover lots of scenarios:
>>
>> 1) Call the Events API with querystring parameters
>> *since=2019-03-20T22:33:44Z&numericDates=false&maxCount=10*
>> 2) Assign the response to variable *previousResponse*
>> 3) Display on the console the items in *previousResponse.data*
>> 4) Start a *while loop* where the criteria to keep running is
>> *previousResponse.moreAvailable* must be true
>> 5) Set to variable *nextStreamPosition* the value in
>> *previousResponse.nextStreamPosition*
>> 6) Call Events API with querystring* streamPosition=<variable
>> nextStreamPosition>&numericDates=false&maxCount=10*
>> 7) Assign the response to variable *previousResponse*
>> 8) Display on the console the items in *previousResponse.data*
>> 9) Repeate loop
>>
>> Thank you,
>>
>> Paulo Lisboa
>> Senior Software Development Engineer
>> paulo.lisboa@smartsheet.com<mailto:paulo.lisboa@smartsheet.com>
>>
>>
>>
>>
>> On Wed, Apr 10, 2019 at 10:38 AM Kevin Fansler <notifications@github.com<mailto:notifications@github.com>>
>> wrote:
>>
>>> @kfansler <https://github.com/kfansler<https://protect-us.mimecast.com/s/XVHbCpYRqGFnYvX8IPa-L4>> requested your review on: #76
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76<https://protect-us.mimecast.com/s/ZI5XCqxVrJt8qzm5hXqPqQ>>
>>> Add enterprise events endpoint .
>>>
>>> —
>>> You are receiving this because your review was requested.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/smartsheet-platform/smartsheet-javascript-sdk/pull/76#event-2266858522<https://protect-us.mimecast.com/s/st2RCrkEvKs8g1Lvh4EAeQ>>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe-auth/AI2I0gRwkt_4vF9bGuTVWtdfm_yR3Zdgks5vfiF6gaJpZM4ciw0d<https://protect-us.mimecast.com/s/r3CSCv2YzOf7Q2M6UAAIkq>>
>>> .
>>>
>>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://protect-us.mimecast.com/s/aJgFCwpEAPUGBAY1C1eZvP>, or mute the thread<https://protect-us.mimecast.com/s/pOUgCxkVBQs1Y9rACWxZjF>.



**This message originated from outside Pariveda Solutions**
